spec/lib/web_of_science/process_records_spec.rb

```diff
@@ -64,9 +64,43 @@
     it 'creates new WebOfScienceSourceRecords' do
       expect { processor.execute }.to change { WebOfScienceSourceRecord.count }
     end
-    it 'creates new Publications'
-    it 'creates new PublicationIdentifiers'
-    it 'creates new Contributions'
+    it 'creates new Publications' do
+      expect { processor.execute }.to change { Publication.count }
+    end
+    it 'creates new PublicationIdentifiers' do
+      expect { processor.execute }.to change { PublicationIdentifier.count }
+    end
+
+    it 'creates new Contributions' do
+      expect { processor.execute }.to change { Contribution.count }
+    end
+    it 'creates new contribution in the pub_hash[:authorship]' do
+      uids = processor.execute
+      pub = Publication.for_uid(uids.sample)
```

Comment:
Using `.sample` in test element selection makes the test suite unrepeatable and non-deterministic.  So even if it fails, you won't know how to repeat/fix that case.  If you want to test all the various objects, then test all of them.  Otherwise, pick any representative slice you want, but don't make it randomized per execution.  

This is not comparable to rspec's execution **order** being random.  That is fine.  The whole point of that is to emphasize how deterministic and isolated each example is.  This approach with `.sample` is the opposite.

Reply:
Literally any one of these publications is representative, so using `sample` is fine here, it gets a representative Publication.  All of the Publication/Contribution models are known quantities.  All of the `uids` have been processed successfully, so any one of them is fine to test here.  No changes required.